app/js/arethusa.search/services/search.js

```diff
@@ -40,6 +40,182 @@ angular.module('arethusa.search').service('search', [
       });
     };
 
+    this.findWordInContext = function(query) {
+
+      return arethusaUtil.inject([], self.strings, function (memo, string, ids) {
+        var matches = []
+        angular.forEach(ids, function(id) {
+          var previousToken = state.getPreviousTokens(id,1);
+          var nextToken = state.getNextTokens(id,1);
+          var nextString = nextToken.length > 0 ? nextToken[0].string : null;
+          var prevString = previousToken.length > 0 ? previousToken[0].string : null;
+          var match = self.compareWordsWithContext(string,prevString,nextString,query.word);
+          if (match.match) {
+            var matchData = {
+              id: id,
+              matchedPrefix: 0,
+              matchedSuffix: 0
+            }
+            if (match.combine < 0) {
+              matchData.includePrevious = previousToken[0].id;
+            } else if (match.combine > 0) {
+              matchData.includeNext = nextToken[0].id;
+            }
+            matches.push(matchData)
+          }
+        });
+        // we only need to try to further narrow by context if we got more than
+        // one hit
+        if (matches.length > 1) {
+          matches.forEach(function (match) {
+            var startingId = match.includePrefix ? match.includePrefix :
+              match.includeSuffix ? match.includeSuffix : match.id;
```

Comment:
I don't think `match.includePrefix` or `match. includeSuffix` is being set anywhere. Should these be `includePrevious`/`includeNext`?

Reply:
ah, thanks yes.